CRM/Core/BAO/CustomGroup.php

```diff
@@ -663,6 +632,35 @@ public static function &getTree(
   }
 
   /**
+   * Clean and validate the filter before it is used in a db query.
+   *
+   * @param string $entityType
+   * @param string $subType
+   *
+   * @return string
+   * @throws \CRM_Core_Exception
+   * @throws \CiviCRM_API3_Exception
+   */
+  protected static function validateSubTypeByEntity($entityType, $subType) {
+    $subType = trim($subType, CRM_Core_DAO::VALUE_SEPARATOR);
```

Comment:
That commasep format is coming from uses like this -

* [`CRM_Contact_Form_Search_Custom_MultipleValues::__construct()`](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/Form/Search/Custom/MultipleValues.php#L50)

That's the only place I found where I see a comma separated value passed in as $entityType, and I feel it ought to be an array so we can not have to handle this weird case. I haven't checked ALL the getTree() calls yet but have done most.

It probably can work, but it feels like something that should be handled by a standard SQL query generator instead of accepting a chunk of SQL from another function.

(moved comment)